umh-core/pkg/service/connection/connection.go

```diff
@@ -0,0 +1,510 @@
+func (c *ConnectionService) convertToNmapConfig(
+	cfg *connectionconfig.ConnectionServiceConfig,
+) nmapserviceconfig.NmapServiceConfig {
+	return nmapserviceconfig.NmapServiceConfig{
+		Target: cfg.Hostname,
+		Port:   int(cfg.Port),
+		// Protocol is not directly mapped in NmapServiceConfig
+		// but could be used to influence scan type
+	}
+}
+
+// updateRecentScans adds a new scan result to the history for flakiness detection.
+// It maintains a circular buffer of recent scan results for each connection,
+// discarding the oldest result when the buffer is full.
+//
+// This history is used by isConnectionFlaky to detect unstable connections
+// that alternate between up and down states.
+func (c *ConnectionService) updateRecentScans(connName string, scan nmap.ServiceInfo) {
+	// Initialize if this is the first scan for this connection
+	if _, exists := c.recentScans[connName]; !exists {
+		c.recentScans[connName] = make([]nmap.ServiceInfo, 0, c.maxRecentScans)
+	}
+
+	// Create a deep copy of the scan info to prevent shared pointers
+	scanCopy := nmap.ServiceInfo{
+		S6FSMState: scan.S6FSMState,
+	}
+
+	// Deep copy the NmapStatus
+	if scan.NmapStatus.LastScan != nil {
+		scanCopy.NmapStatus.IsRunning = scan.NmapStatus.IsRunning
+		scanCopy.NmapStatus.LastScan = &nmap.NmapScanResult{
+			Timestamp: scan.NmapStatus.LastScan.Timestamp,
+			Error:     scan.NmapStatus.LastScan.Error,
+			PortResult: nmap.PortResult{
+				Port:      scan.NmapStatus.LastScan.PortResult.Port,
+				State:     scan.NmapStatus.LastScan.PortResult.State,
+				LatencyMs: scan.NmapStatus.LastScan.PortResult.LatencyMs,
+			},
+			Metrics: scan.NmapStatus.LastScan.Metrics,
+		}
+	}
+
+	// Add the new scan (keeping only the most recent ones)
+	scans := c.recentScans[connName]
+	if len(scans) >= c.maxRecentScans {
+		// Remove oldest scan
+		scans = scans[1:]
+	}
+	scans = append(scans, scanCopy)
+	c.recentScans[connName] = scans
+}
+
+// isConnectionFlaky determines if a connection is flaky based on recent scan history.
+// A connection is considered flaky if it has shown different port states
+// in the recent history window (default: last 5 scans).
+//
+// The algorithm looks for:
+// - Mix of open and closed states
+// - Mix of open and filtered states
+// - Mix of closed and filtered states
+//
+// At least 3 samples are required to make this determination, to avoid
+// false positives from normal temporary network conditions.
+func (c *ConnectionService) isConnectionFlaky(connName string) bool {
+	scans, exists := c.recentScans[connName]
+	if !exists || len(scans) < 3 {
+		// Need at least 3 samples to determine flakiness
+		return false
+	}
+
+	// Check for changes in port state in last scans
+	hasOpen := false
+	hasClosed := false
+	hasFiltered := false
+
+	for _, scan := range scans {
+		if scan.NmapStatus.LastScan == nil {
+			continue
+		}
+
+		state := scan.NmapStatus.LastScan.PortResult.State
+		switch state {
+		case "open":
```

Comment:
These would be nice enums

Reply:
i think this is more of a style thing, and we already merged nmap, so let me stay here consistent with nmap